source/internal/kernelutils/kernel_kconfig.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2018-2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kernelutils
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/node-feature-discovery/source"
+)
+
+// Read gzipped kernel config
+func readKconfigGzip(filename string) ([]byte, error) {
+	// Open file for reading
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	// Uncompress data
+	r, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	return ioutil.ReadAll(r)
+}
+
+// Read kconfig into a map
+func ParseKconfig(searchPaths []string) (map[string]string, error) {
```

Comment:
nit: it'd be easier to review if moving the code and changing the functionality was done in separate commits

Reply:
fixed